docs/guides/01_creating_an_aggregate.md

```diff
@@ -0,0 +1,190 @@
+
+Let's follow with a class representing a task belonging to a project:
+
+```csharp
+public class Task : IIdentifiable<SId<Task>>
+{
+    public SId<Task> Id { get; private init; }
+    public string Name { get; private set; }
+    public TaskStatus Status { get; private set; }
+    public SId<User>? AssignedUser { get; private set; }
+
+    public Project ParentProject { get; private init; }
+
+    DateTime IOptimisticConcurrency.DateModified { get; set; }
+
+    private Task() { }
+
+    public static Task Create(Project parentProject, string name)
+    {
+        return new Task
+        {
+            Name = name,
+            ParentProject = parentProject,
+            Status = TaskStatus.NotStarted,
+        };
+    }
+
+    public enum TaskStatus
+    {
+        NotStarted,
+        InProgress,
+        Finished,
+    }
+}
+```
+
+Notice that a task is not an aggregate root - it can only be accessed through the project to which it belongs.
+
+Likewise, let's create a class representing an user which can be assigned to a task:
+
+```csharp
+public class User : IAggregateRoot<SId<User>>
+{
+    public SId<User> Id { get; private init; }
+    public string Name { get; private set; }
+    public string Email { get; private set; }
+
+    DateTime IOptimisticConcurrency.DateModified { get; set; }
+
+    private Task() { }
+
+    public static Task Create(string name, string email)
+    {
+        return new Task
+        {
+            Id = SId<User>.New(),
+            Name = name,
+            Email = email,
+        };
+    }
+}
+```
+
+Notice that the `User` class is an aggregate root - the user represents a standalone business domain object with its own meaning and behavior.
+
+## Adding logic
+
+So far, only the structure of objects has been defined but business domain objects usually has some set of behaviors which they can perform. In this section we will see how a logic can be added to the aggregates.
+
+Let's start with a simple example of adding a few methods to the `Task` class:
+
+```csharp
+public class Task : IIdentifiable<SId<Task>>
+{
+    . . .
+
+    public void Edit(string name)
+    {
+        Name = name;
+    }
+
+    public void AssignUser(SId<User> userId)
+    {
+        AssignedUser = userId;
+    }
+
+    public void UnassignUser()
+    {
+        AssignUser = null;
+    }
+
+    public void ChangeStatus(TaskStatus status)
+    {
+        Status = status;
+    }
+}
+```
+
+Remember that the only way to access a task is to go through the relevant project which is the aggregate root. Because of this we will add some methods to the `Project` class which will allow us to interact with the tasks:
+
+```csharp
+public class Project : IIdentifiable<SId<Project>>
+{
+    . . .
+
+    public void EditTask(SId<Task> taskId, string name)
+    {
+        tasks.Single(t => t.Id == taskId).Edit(name);
```

Comment:
I don't ever use `.Single(...)` in my projects, is a mistake by me, here or it doesn't matter?

Reply:
I would say it doesn't matter that much. I find `Single(...)` more meaningful and in this this case it makes it less likely for two tasks with the same `Id` to be left unnoticed. But of course it is very unlikely in the first place.